src/Bicep.Core/Analyzers/Linter/Rules/BCPL1070.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Analyzers.Interfaces;
+using Bicep.Core.CodeAction;
+using Bicep.Core.Parsing;
+using Bicep.Core.Semantics;
+using Bicep.Core.Syntax;
+using Bicep.Core.Visitors;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Bicep.Core.Analyzers.Linter.Rules
+{
+    public sealed class BCPL1070 : LinterRule
+    {
+        public BCPL1070() : base(
+            code: "BCPL1070",
+            ruleName: "Unecessary dependsOn",
+            description: "Best Practice: remove unnecessary dependsOn.",
+            docUri: "https://bicep/linter/rules/BCPL1070") // TODO: setup up doc pages
+        { }
+
+        override public IEnumerable<IBicepAnalyzerDiagnostic> Analyze(SemanticModel model)
+        {
+            var spanDiagnostics = new List<TextSpan>();
+            var visitor = new BCPL1070Visitor(model, spanDiagnostics);
+            visitor.Visit(model.SyntaxTree.ProgramSyntax);
+            return spanDiagnostics.Select(span => CreateDiagnosticForSpan(span));
+        }
+
+        private sealed class BCPL1070Visitor : SyntaxVisitor
+        {
+            private readonly SemanticModel semanticModel;
+            private readonly List<TextSpan> diagnostics;
+
+            public BCPL1070Visitor(SemanticModel semanticModel, List<TextSpan> diagnostics)
+            {
+                this.semanticModel = semanticModel;
+                this.diagnostics = diagnostics;
+            }
+
+            public override void VisitObjectPropertySyntax(ObjectPropertySyntax syntax)
+            {
+                if (syntax.NameEquals("dependsOn"))
+                {
+                    if (syntax.Value is ArraySyntax dependencies)
```

Comment:
>dependencies [](http://example.com/codeflow?start=52&length=12)

How does this deal with a property called "dependsOn" whose value is an array of strings in a variable, for example?

Reply:
Or if someone put the dependsOn at the wrong level in the resource? Would we still analyze it?

Reply:
Removed this rule for now.